### ISSUE TITLE
Same fields in different content types get overwritten

### ISSUE THREAD

**Issue by oriolroger** [backdrop-contrib/corresponding_node_references#6]
Hi all, I'm facing a problem and not sure if it's a bug in the admin section of this module.

I have two content types with two node_reference fields, one pointing to another one. If I go to the admin page of this module (/admin/config/content/corresponding_node_references) I can see and enable the relationship between them. All ok.

_Content type 1   <->   Reference to Content type 3_

But if I add one of these same fields to a third content type, I'd expect to see two relationships in the admin page, and instead I don't see anything. 

_Content type 2   <->   Reference to Content type 3 (NOT WORKING)_

I think the code is searching for field instances of node_reference fields, and once it find one it's overwritting the latest one, instead of adding a new one.

I've patched the file "_corresponding_node_references.admin.inc_" in my site with these new lines, hopefully that can help somebody, and probably get a new version of this module.

```
function corresponding_node_references_settings_form() {
  ...
  $instance_config_names = config_get_names_with_prefix('field.instance.node.');
  $instances = array();
  foreach ($instance_config_names as $instance_config_name) {
    $config_instance = config($instance_config_name);
    $field_name = $config_instance->get('field_name');
    $instances[] = array(
      'field_name' => $field_name,
      'entity_type' => $config_instance->get('entity_type'),
      'bundle' => $config_instance->get('bundle'),
    );
  }
  ...
}
```

Any feedback would be appreciated. Thanks!

**Comment by argiepiano** [comment]
I don't have any experience with Corresponding Node References, but I did use Corresponding Entity References (CER) in Drupal 7, which I believe works similarly. 

Adding those reference fields to another bundle does not create a new relationship. In CER, the actual relationship definition is saved as a configuration entity in its own table, and can only be created manually through the UI (or API).

But perhaps CNR is different from CER?

**Comment by oriolroger** [comment]
Thanks @argiepiano, I was expecting CNR to be working the same way, but it's not. It seems that the module is automatically creating relationships when it finds two fields referencing each other, and then it displays them in the UI, where the admin user can enable/disable those relationships. The settings for enabled/disabled relationships are then saved as configuration files, like a usual Backdrop config.

**Comment by herbdool** [comment]
I think this can be closed.

**Comment by argiepiano** [comment]
Yes, it was the same issue I patched yesterday. I wish I had looked at this! This had a somewhat similar solution for the same problem.
